test/NosCore.Tests/HandlerTests/InventoryPacketControllerTests.cs

```diff
@@ -888,5 +881,25 @@ public void Test_Transform_Delay()
             var packet = (DelayPacket)_session.LastPacket;
             Assert.IsTrue(packet.Delay == 5000);
         }
+
+        [TestMethod]
+        public void Test_Increment_SpAdditionPoints()
+        {
+            _session.Character.SpAdditionPoint = 0;
+            _session.Character.Inventory.AddItemToPocket(_itemBuilder.Create(1078, 1));
+            var item = _session.Character.Inventory.First();
+            _handler.UseItem(new UseItemPacket { VisualType = _session.Character.VisualType, VisualId = _session.Character.VisualId, Type = item.Value.Type, Slot = item.Value.Slot, Mode = 0, Parameter = 0 });
+            Assert.IsTrue(_session.Character.SpAdditionPoint != 0);
+        }
+
+        [TestMethod]
+        public void Test_Overflow_SpAdditionPoints()
+        {
+            _session.Character.SpAdditionPoint = _session.WorldConfiguration.MaxAdditionalSpPoints;
+            _session.Character.Inventory.AddItemToPocket(_itemBuilder.Create(1078, 1));
+            var item = _session.Character.Inventory.First();
+            _handler.UseItem(new UseItemPacket { VisualType = _session.Character.VisualType, VisualId = _session.Character.VisualId, Type = item.Value.Type, Slot = item.Value.Slot, Mode = 0, Parameter = 0 });
+            Assert.IsTrue(_session.Character.SpAdditionPoint == _session.WorldConfiguration.MaxAdditionalSpPoints && _session.LastPacket is MsgPacket);
```

Comment:
You need to test the message value both for the case you have max and use item and a test with max-1 and use item